provisioning/miniprov/hare_mp/main.py

```diff
@@ -288,60 +279,83 @@ def save(filename: str, contents: str) -> None:
-                   default='',
-                   dest='config_url',
-                   action='store')
+    subparser = p.add_subparsers()
+
+    parser = subparser.add_parser('post_install',
+                                  help='Perform post installation checks')
+    parser.set_defaults(func=post_install)
+    parser.add_argument(
+        '--report-unavailable-features',
+        help='Report unsupported features according to setup type',
+        action='store_true')
+    parser.add_argument('--configure-logrotate',
+                        help='Configure logrotate for hare',
+                        action='store_true')
+
+    parser = subparser.add_parser('config', help='Configure Hare')
+    parser.set_defaults(func=config)
+    parser.add_argument('--config',
+                        help='Conf Store URL with cluster info',
+                        required=True,
+                        nargs=1,
+                        type=str,
+                        action='store')
+    parser.add_argument(
+        '--file',
+        help='Full path to the CDF file to generate at this stage.',
+        nargs=1,
+        default=['/var/lib/hare/cluster.yaml'],
+        type=str,
+        action='store')
+
+    parser = subparser.add_parser('init',
+                                  help='Perform component initialization')
+    parser.set_defaults(func=init)
+    parser.add_argument('--file',
+                        help='Full path to the CDF file.',
+                        nargs=1,
+                        default=['/var/lib/hare/cluster.yaml'],
+                        type=str,
+                        action='store')
+
+    parser = subparser.add_parser('test', help='Testing cluster status')
+    parser.set_defaults(func=test)
+    parser.add_argument('--file',
+                        help='Full path to the CDF file.',
+                        nargs=1,
+                        default=['/var/lib/hare/cluster.yaml'],
+                        type=str,
+                        action='store')
+
+    parser = subparser.add_parser('support_bundle',
+                                  help='Generating support bundle')
+    parser.set_defaults(func=generate_support_bundle)
+
+    parser = subparser.add_parser('reset', help='Reset/cleanup step')
+    parser.set_defaults(func=cleanup)
+    parser = subparser.add_parser('cleanup', help='Reset/cleanup step')
+    parser.set_defaults(func=cleanup)
+
     setup_logging()
 
     parsed = p.parse_args(sys.argv[1:])
-    if parsed.config_url:
-        # --config also depends on optional parameter --filename,
-        # that is why the approach with putting the whole confi
-        # logic into the ConfigAction class will not work.
-        url = parsed.config_url[0]
-        filename = parsed.filename[0] or '/var/lib/hare/cluster.yaml'
-        save(filename, generate_cdf(url))
+
+    if not hasattr(parsed, 'func'):
+        logging.error('Error: No valid command passed. Please check "--help"')
```

Comment:
[suggestion]: `logging.error('Error: Invalid command, please check hare_setup --help')`. 
Also it would be good if we can simply print help.